front/app/containers/Admin/formBuilder/components/FormFields/FieldRuleDisplay/index.tsx

```diff
@@ -20,14 +30,35 @@ export const FieldRuleDisplay = ({
         height="24px"
         data-cy="e2e-field-rule-display"
       >
-        <Icon fill={colors.coolGrey500} width="18px" name="logic" my="auto" />
+        <Icon
+          fill={isRuleValid ? colors.coolGrey500 : colors.error}
+          width="18px"
+          name="logic"
+          my="auto"
+        />
         <Text my="auto" pl="8px" pr="4px" color="coolGrey600" fontSize="s">
           {answerTitle}
         </Text>
-        <Icon fill={colors.teal300} width="16px" name="arrow-right" my="auto" />
+        <Icon
+          fill={isRuleValid ? colors.teal300 : colors.error}
+          width="16px"
+          name="arrow-right"
+          my="auto"
+        />
         <Text my="auto" px="4px" color="coolGrey600" fontSize="s">
           {targetPage}
         </Text>
+        {!isRuleValid && (
+          <Box my="auto" ml="8px">
+            <Badge
+              style={{ padding: '2px 6px 2px 6px' }}
```

Comment:
The padding here isn't great, but the component from the library is too large for this row.